content/draft/general/password_strength.md

```diff
@@ -0,0 +1,91 @@
+
+## Designs
+
+* [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=11%3A4921)
+
+## Use Cases
+
+### Account creation
+
+1. User types the choosen password in the password field
+2. System displays feedback on the strength of the password
+2.1 System checks if the password fits criteria (see Functional Requirements#password criteria) and displays feedback (e.g "Password is too short")
+2.2 System categorizes password strength (see Functional Requirements#password categorization) and displays matching category (e.g password Is "Good")
+2.2 System checks if password is a common password (see Functional Requirements#password validation) and displays feedback (e.g "this pasword has been pwned and shouldn't be used")
+3. User types password again in the confirm password field
+4. System checks if password matches and displays "Passwords don't match" if they don't
+5. System checks if password matches and enables the "Create" button if they do
+6. User clicks "Create" button
+
+### Change password
+
+1. User types the current password in the current password field
+2. User types the choosen password in the new password field
+3. System displays feedback on the strength of the password
+4.1 System checks if the password fits criteria (see Functional Requirements#password criteria) and displays feedback (e.g "Password is too short")
+4.2 System categorizes password strength (see Functional Requirements#password categorization) and displays matching category (e.g password Is "Good")
+4.2 System checks if password is a common password (see Functional Requirements#password validation) and displays feedback (e.g "this pasword has been pwned and shouldn't be used")
+5. User types password again in the confirm password field
+6. System checks if password matches and displays "Passwords don't match" if they don't
+7. System checks if password matches and enables the "Create" button if they do
+8. User clicks "Create" button
+7. System checks if current password is valid and displays "Current password is invalid" if it isn't
+
+## Functional Requirements
+
+### Password criteria
+- A password MUST BE at least 6 characters long.
+- A password MUST BE at most 64 characters long.
+- A password MUST NOT contain spaces.
+- A password MAY CONTAIN any printable ascii character.
+
+### Password validation
+
+- A password MUST be checked against a list of common passwords
+    - If a password matches a common password, it MUST BE rejected
+    - If a password matches a common password, it MUST BE display a warning "this pasword has been pwned and shouldn't be used"
+    - If the application is online, the password MAY BE checked against an online or a local database.
+    - If the application is offline, the password MAY BE checked against a local database.
```

Comment:
What is that local database? What are "common" passwords? Is this out of band? Wonder how we can be spec compliant if we don't know what database of common passwords we should refer to...

Reply:
I don't think we should worry about performing this check if the application is offline, because these databases are huge.  IMHO not worth the increase in app size, or the user bandwidth to download these databases after the app is installed.  So if the application is offline, I think we can skip the 'has this password been pwned' check.